src/RunTests/RunVSTestTask.cs

```diff
@@ -0,0 +1,443 @@
+
+        /// <summary>
+        /// Gets or Sets Verbosity level of VSTest output.
+        /// </summary>
+        public string VSTestVerbosity { get; set; }
+
+        /// <summary>
+        /// Gets or Sets Collectors for VSTest run.
+        /// </summary>
+        public string[] VSTestCollect { get; set; }
+
+        /// <summary>
+        /// Gets or Sets source blame on test failure.
+        /// </summary>
+        public string VSTestBlame { get; set; }
+
+        /// <summary>
+        /// Gets or Sets source blame on test crash.
+        /// </summary>
+        public string VSTestBlameCrash { get; set; }
+
+        /// <summary>
+        /// Gets or Sets Dumptype used for crash source blame.
+        /// </summary>
+        public string VSTestBlameCrashDumpType { get; set; }
+
+        /// <summary>
+        /// Gets or Sets source blame on test crash even if test pass.
+        /// </summary>
+        public string VSTestBlameCrashCollectAlways { get; set; }
+
+        /// <summary>
+        /// Gets or Sets source blame on test hang.
+        /// </summary>
+        public string VSTestBlameHang { get; set; }
+
+        /// <summary>
+        /// Gets or Sets Dumptype used for hang source blame.
+        /// </summary>
+        public string VSTestBlameHangDumpType { get; set; }
+
+        /// <summary>
+        /// Gets or Sets Time out for hang source blame.
+        /// </summary>
+        public string VSTestBlameHangTimeout { get; set; }
+
+        /// <summary>
+        /// Gets or Sets The directory path where trace data collector is.
+        /// </summary>
+        public string VSTestTraceDataCollectorDirectoryPath { get; set; }
+
+        /// <summary>
+        /// Gets or Sets disabling Microsoft logo while running test through VSTest.
+        /// </summary>
+        public string VSTestNoLogo { get; set; }
+
+        /// <summary>
+        /// Gets or Sets Test artifacts processing mode which is applicable for .NET 5.0 or later versions.
+        /// </summary>
+        public string VSTestArtifactsProcessingMode { get; set; }
+
+        /// <summary>
+        /// Gets or Sets Correlation Id of test session.
+        /// </summary>
+        public string VSTestSessionCorrelationId { get; set; }
+
+        /// <summary>
+        /// Gets or Sets Runner version of VSTest.
+        /// </summary>
+        public string VSTestRunnerVersion { get; set; }
+
+        /// <summary>
+        /// Gets or Sets Path to nuget package cache.
+        /// </summary>
+        public string NugetPath { get; set; }
+
+        /// <summary>
+        /// Executes the test. Skips execution if specified.
+        /// </summary>
+        /// <returns>Returns true if the test was executed, otherwise false.</returns>
+        public override bool Execute()
+        {
+            var traceEnabledValue = Environment.GetEnvironmentVariable("VSTEST_BUILD_TRACE");
+            if (SkipExecution)
```

Comment:
Should this be skipped MSBuild on the Target instead of invoking the task just to short-circuit immediately?